scripts/ASTImportTest.sh

```diff
@@ -0,0 +1,105 @@
+SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-build}
+SOLC=${REPO_ROOT}/${SOLIDITY_BUILD_DIR}/solc/solc
+SPLITSOURCES=${REPO_ROOT}/scripts/splitSources.py
+
+SYNTAXTESTS_DIR="${REPO_ROOT}/test/libsolidity/syntaxTests"
+NSOURCES="$(find $SYNTAXTESTS_DIR -type f | wc -l)"
+
+# DEV_DIR="${REPO_ROOT}/../tmp/contracts/"
+# NSOURCES="$(find $DEV_DIR -type f | wc -l)" #TODO use find command
+
+FAILED=0
+UNCOMPILABLE=0
+TESTED=0
+
+# function tests whether exporting and importing again leaves the JSON ast unchanged
+# Results are recorded by adding to FAILED or UNCOMPILABLE.
+# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Expected parameters:
+# $1 name of the file to be exported and imported
+# $2 any files needed to do so that might be in parent directories
+function testImportExportEquivalence {
+    if $SOLC $1 $2 > /dev/null 2>&1
+    then
+        # save exported json as expected result (silently)
+        $SOLC --combined-json ast,compact-format --pretty-json $1 $2> expected.json 2> /dev/null
+        # import it, and export it again as obtained result (silently)
+        $SOLC --import-ast --combined-json ast,compact-format --pretty-json expected.json > obtained.json 2> /dev/null
+        if [ $? -ne 0 ]
+           then
+               # For investigating, use exit 1 here so the script stops at the
+               # first failing test
+               # exit 1
+               FAILED=$((FAILED + 1))
+               return 1
+        fi
+        DIFF="$(diff expected.json obtained.json)"
+        if [ "$DIFF" != "" ]
+        then
+            if [ "$DIFFVIEW" == "" ]
+            then
+                echo -e "ERROR: JSONS differ for $1: \n $DIFF \n"
+                echo "Expected:"
+                echo "$(cat ./expected.json)"
+                echo "Obtained:"
+                echo "$(cat ./obtained.json)"
+            else
+                # Use user supplied diff view binary
+                $DIFFVIEW expected.json obtained.json
+            fi
+            FAILED=$((FAILED + 1))
+            return 2
+        fi
+        TESTED=$((TESTED + 1))
+        rm expected.json obtained.json
+    else
+        # echo "contract $solfile could not be compiled "
+        UNCOMPILABLE=$((UNCOMPILABLE + 1))
+    fi
+    # return 0
+}
+echo "Looking at $NSOURCES .sol files..."
+
+# for solfile in $(find $DEV_DIR -name *.sol)
+for solfile in $(find $SYNTAXTESTS_DIR -name *.sol)
+do
+    OUTPUT=$($SPLITSOURCES $solfile)
+    if [ $? != 1 ]
+    then
+        # echo $OUTPUT
+        NSOURCES=$((NSOURCES - 1))
+        for i in $OUTPUT;
+        do
+            testImportExportEquivalence $i $OUTPUT
+            NSOURCES=$((NSOURCES + 1))
+        done
+        # delete created files
+        rm -r $OUTPUT
+        for f in $OUTPUT:
+        do
+            # remove everything after the first '/' to delete all
+            # subdirectories and content
+            echo $f | cut -d'/' -f-1 | xargs -d"\n" rm -r 2> /dev/null
```

Comment:
Strong objection against this.
We should create a temporary directory at the very beginning and remove that.

Reply:
That's what we're doing: https://github.com/ethereum/solidity/pull/7153/files#diff-a8a5760144e03e410389fb0afd20e8c8R440
This is only doing cleanup within that temporary directory for the next test.

Reply:
but we could create another tmp directory so we don't use `rm` in such an intransparent and potentially dangerous way.

Reply:
yes, maybe that would be cleaner